java/cuvs-java/src/main/java22/com/nvidia/cuvs/internal/CuVSResourcesImpl.java

```diff
@@ -65,11 +61,25 @@ public void close() {
     synchronized (this) {
       int returnValue = cuvsResourcesDestroy(resourceHandle);
       checkCuVSError(returnValue, "cuvsResourcesDestroy");
+      hostBuffer.close();
     }
   }
 
   @Override
   public Path tempDirectory() {
     return tempDirectory;
   }
+
+  public static MemorySegment getHostBuffer(ScopedAccess access) {
+
+    while (access instanceof DelegatingScopedAccess delegatingScopedAccess) {
```

Comment:
TIL.  I wasn't aware of this idiom.

Reply:
This whole pattern is inspired by Lucene, which has the same problem of not exposing some implementation classes (e.g. pre-Java 22 FFI classes) externally, but the need to access them internally. So we have a "Access" interface (that we can inspect and cast to) that tags a class with a capability (in this case, the presence of a memory segment that can be used as a buffer).